ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -96,56 +94,66 @@ import           Development.IDE.Types.Location
 import           Development.IDE.Types.Options
 import           GHC                               (ForeignHValue,
                                                     GetDocsFailure (..),
-                                                    GhcException (..),
                                                     parsedSource)
 import qualified GHC.LanguageExtensions            as LangExt
 import           GHC.Serialized
-import           HieDb
+import           HieDb                             hiding (withHieDb)
 import qualified Language.LSP.Server               as LSP
 import           Language.LSP.Protocol.Types                (DiagnosticTag (..))
 import qualified Language.LSP.Protocol.Types                as LSP
 import qualified Language.LSP.Protocol.Message            as LSP
 import           System.Directory
 import           System.FilePath
 import           System.IO.Extra                   (fixIO, newTempFileWithin)
-import           Unsafe.Coerce
+
+
+#if !MIN_VERSION_ghc(9,0,1)
+import           HscTypes
+import           TcSplice
+#endif
 
 #if MIN_VERSION_ghc(9,0,1)
 import           GHC.Tc.Gen.Splice
+#endif
+
+#if MIN_VERSION_ghc(9,0,1) && !MIN_VERSION_ghc(9,2,1)
```

Comment:
I believe `else` CPP is not worth it in the imports section. As this statement uses an `&&` symbol, it would need to nested CPP, and then you're pretty much back to where we started.